packages/database/src/databases/provider.ts

```diff
@@ -63,53 +65,99 @@ import {
 	type UserSolutionRecord,
 	UserSolutionRecordSchema,
 } from "@prosopo/types-database";
-import { type DeleteResult, ServerApiVersion } from "mongodb";
-import mongoose, { type Connection, type ObjectId } from "mongoose";
-
-mongoose.set("strictQuery", false);
-
-// mongodb://username:password@127.0.0.1:27017
-const DEFAULT_ENDPOINT = "mongodb://127.0.0.1:27017";
-
-/**
- * Returns the Database object through which Providers can put and get captchas
- * @param {string} url          The database endpoint
- * @param {string} dbname       The database name
- * @return {ProsopoDatabase}    Database layer
- */
-export class ProsopoDatabase extends AsyncFactory implements Database {
-	url: string;
-	tables?: Tables;
-	dbname: string;
-	connection?: Connection;
-	logger: Logger;
-
-	constructor() {
-		super();
-		this.url = "";
-		this.dbname = "";
-		this.logger = getLoggerDefault();
-	}
+import type { DeleteResult } from "mongodb";
+import type { ObjectId } from "mongoose";
+import { MongoDatabase } from "../base/mongo.js";
+
+enum TableNames {
+	captcha = "captcha",
+	dataset = "dataset",
+	solution = "solution",
+	commitment = "commitment",
+	usersolution = "usersolution",
+	pending = "pending",
+	scheduler = "scheduler",
+	powcaptcha = "powcaptcha",
+	client = "client",
+}
 
-	public async init(
+const PROVIDER_TABLES = [
+	{
+		collectionName: TableNames.captcha,
+		modelName: "Captcha",
+		schema: CaptchaRecordSchema,
+	},
+	{
+		collectionName: TableNames.powcaptcha,
+		modelName: "PowCaptcha",
+		schema: PoWCaptchaRecordSchema,
+	},
+	{
+		collectionName: TableNames.dataset,
+		modelName: "Dataset",
+		schema: DatasetRecordSchema,
+	},
+	{
+		collectionName: TableNames.solution,
+		modelName: "Solution",
+		schema: SolutionRecordSchema,
+	},
+	{
+		collectionName: TableNames.commitment,
+		modelName: "UserCommitment",
+		schema: UserCommitmentRecordSchema,
+	},
+	{
+		collectionName: TableNames.usersolution,
+		modelName: "UserSolution",
+		schema: UserSolutionRecordSchema,
+	},
+	{
+		collectionName: TableNames.pending,
+		modelName: "Pending",
+		schema: PendingRecordSchema,
+	},
+	{
+		collectionName: TableNames.scheduler,
+		modelName: "Scheduler",
+		schema: ScheduledTaskRecordSchema,
+	},
+	{
+		collectionName: TableNames.client,
+		modelName: "Client",
+		schema: ClientRecordSchema,
+	},
+];
+
+export class ProviderDatabase extends MongoDatabase implements IDatabase {
+	tables = {} as Tables<TableNames>;
+
+	constructor(
 		url: string,
-		dbname: string,
-		logger: Logger,
+		dbname?: string,
```

Comment:
why is this optional? surely you need a dbname?

Reply:
you can pass the dbname in the URI so its not needed here